server/src/main/java/io/druid/query/LocatedSegmentDescriptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.query;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import io.druid.server.coordination.DruidServerMetadata;
+import org.joda.time.Interval;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ */
+public class LocatedSegmentDescriptor extends SegmentDescriptor
+{
+  private final long size;
+  private final List<DruidServerMetadata> locations;
+
+  @JsonCreator
+  public LocatedSegmentDescriptor(
+      @JsonProperty("itvl") Interval interval,
```

Comment:
SegmentDescriptor is not exposed to the user so it's less of a problem. However, all our classes that are exposed to the user, use either `interval` or `intervals` so we should keep that consistent.